craft/loopback/src/loopback.scala

```diff
@@ -31,10 +31,11 @@ class Lloopback(c: loopbackParams)(implicit p: Parameters) extends LloopbackBase
 
 }
 
-class NloopbackTop(c: NloopbackTopParams)(implicit p: Parameters) extends NloopbackTopBase(c)(p)
+class NloopbackTop(c: NloopbackTopParams)(implicit p: Parameters) extends NloopbackTopBase(c)(p) with BindingScope
 {
 
 // User code here
+  ResourceBinding { Resource(imp.device, "exists").bind(ResourceString("yes")) }
```

Comment:
@richardxia @sistillson I think this is something we probably want to always emit in the base code?

Reply:
What exactly does this do? I see this [exact line](https://github.com/chipsalliance/rocket-chip/blob/1f561026f85b885a7b13c3a5dffda6a47fa2d6bb/src/main/scala/diplomacy/Resources.scala#L89) in `DeviceSnippet` in rocket-chip.

`DeviceSnippet` itself, which is an abstract class, seems to only be used to define and immediately instantiate an  anonymous class which overrides the `describe()` method. It feels like we should at the very least move this to the base code emitted by duh-scala, but we may also want to use `DeviceSnippet` itself.